### PR TITLE
Script for TCs TEST-37514, TEST-37496, TEST-37359

### PR DESCRIPTION
**Pull request by ashukakkar**
# Problem Statement
- Upgrade feature TC Automation Scripts.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by ashukakkar** [comment]
Please find HTML TC run report and console logs is in below zip file:
[Upgrade TC Automation report.zip](https://github.com/Seagate/cortx-test/files/8313263/Upgrade.TC.Automation.report.zip)

File: comptests/prov/test_prov_k8s_cortx_upgrade.py [hunk 1]
```diff
@@ -269,14 +268,84 @@ def test_32605(self):
...
+    @pytest.mark.comp_prov
+    @pytest.mark.tags("TEST-37514")
+    def test_37514(self):
+        """
+        Verify Upgrade script throws an error message when invalid argument is passed.
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Check proper error message when invalid argument passed to upgrade script.")
+        upgrade_cmd = "sh upgrade-cortx-cloud.sh -abc"
+        cmd = "cd {}; {}".format(self.prov_deploy_cfg["git_remote_path"], upgrade_cmd)
```
Comment (pragamj) [inline comptests/prov/test_prov_k8s_cortx_upgrade.py:294]: can we add this in command.py file

File: comptests/prov/test_prov_k8s_cortx_upgrade.py [hunk 2]
```diff
@@ -269,14 +268,84 @@ def test_32605(self):
...
+    @pytest.mark.tags("TEST-37496")
+    def test_37496(self):
+        """
+        Verify Upgrade script throws an error message when no argument is passed.
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Check proper error message when no argument passed to upgrade script.")
+        upgrade_cmd = "sh upgrade-cortx-cloud.sh"
+        error_msg = "ERROR: Required option POD_TYPE is missing."
+        cmd = "cd {}; {}".format(self.prov_deploy_cfg["git_remote_path"], upgrade_cmd)
```
Comment (pragamj) [inline comptests/prov/test_prov_k8s_cortx_upgrade.py:313]: same as mentioned above.

File: comptests/prov/test_prov_k8s_cortx_upgrade.py (hunk 1 above)
Comment (ashukakkar) [reply comptests/prov/test_prov_k8s_cortx_upgrade.py:294]: Hi @pragamj ,

This is specific to this TC only. Do we want to add in commands.py?

Comment (ashukakkar) [reply comptests/prov/test_prov_k8s_cortx_upgrade.py:294]: Ok. I'll add these to commands.py file.

Comment (ashukakkar) [reply comptests/prov/test_prov_k8s_cortx_upgrade.py:294]: Or else we can add these to prov specific yaml file, i.e., test_prov_config.yaml? Either yaml file or commands.py please suggest.

**Review by priyankaborawake — APPROVED** [review]
looks good to me

**Review by priyankaborawake — COMMENTED** [review]
@ashukakkar Please change the branch to main and not to dev

File: comptests/prov/test_prov_k8s_cortx_upgrade.py (hunk 1 above)
Comment (ashukakkar) [reply comptests/prov/test_prov_k8s_cortx_upgrade.py]: for now I have kept them in test_prov_config.yaml file as these are particular to these TCs only.

File: comptests/prov/test_prov_k8s_cortx_upgrade.py (hunk 2 above)
Comment (ashukakkar) [reply comptests/prov/test_prov_k8s_cortx_upgrade.py]: for now I have kept them in test_prov_config.yaml file as these are particular to these TCs only.

File: comptests/prov/test_prov_k8s_cortx_upgrade.py (hunk 1 above)
Comment (pragamj) [reply comptests/prov/test_prov_k8s_cortx_upgrade.py]: better to use commands.py as we have all deployment and upgrade commands in their.

Comment (ashukakkar) [reply comptests/prov/test_prov_k8s_cortx_upgrade.py]: Done. Added in commands.py file.

File: comptests/prov/test_prov_k8s_cortx_upgrade.py [hunk 3]
```diff
@@ -269,14 +268,82 @@ def test_32605(self):
...
+    @pytest.mark.comp_prov
+    @pytest.mark.tags("TEST-37496")
+    def test_37496(self):
+        """
+        Verify Upgrade script throws an error message when no argument is passed.
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Check proper error message when no argument passed to upgrade script.")
+        error_msg = "ERROR: Required option POD_TYPE is missing."
+        cmd = commands.UPGRADE_NO_PARAM_CMD.format(self.prov_deploy_cfg["git_remote_path"])
```
Comment (pragamj) [inline comptests/prov/test_prov_k8s_cortx_upgrade.py:311]: rename cmd to upgrade_command or some meaning full param

**Review by pragamj — APPROVED** [review]

**Review by sonalk0209 — APPROVED** [review]
Approving for merge.

**Review by ravindrachoudhari — APPROVED** [review]

File: comptests/prov/test_prov_k8s_cortx_upgrade.py [hunk 4]
```diff
@@ -269,14 +268,82 @@ def test_32605(self):
...
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Check proper error message when no argument passed to upgrade script.")
+        error_msg = "ERROR: Required option POD_TYPE is missing."
+        cmd = commands.UPGRADE_NEG_CMD.format(self.prov_deploy_cfg["git_remote_path"])
+        resp = self.master_node_obj.execute_cmd(cmd=cmd, exc=False)
+        if isinstance(resp, bytes):
+            resp = str(resp, 'UTF-8')
+        resp = "".join(resp).replace("\\n", "\n")
+        print(resp)
```
Comment (seagate-sarang-sawant) [inline comptests/prov/test_prov_k8s_cortx_upgrade.py:316]: Can you replace print with log statement ?

Comment (ashukakkar) [reply comptests/prov/test_prov_k8s_cortx_upgrade.py]: Removed. Actually this is not required.

**Review by seagate-sarang-sawant — APPROVED** [review]

**Pull request merged**